### PR TITLE
Escape apostrophe in XML parser, refs #13204

### PR DESCRIPTION
**Pull request by mcantelon**
Apostrophes were causing a syntax error during eval in XML parsing
code.

### CODE REVIEW

File: lib/QubitXmlImport.class.php [hunk 1]
```diff
@@ -658,7 +658,7 @@ private function processMethods(&$domNode, &$importDOM, $methods, &$currentObjec
                         if (null !== ($node = eval('return '.$str.';')))
                         {
                           // Substitute node value for search string
-                          $parameter = str_replace($match, '\''.$node->nodeValue.'\'', $parameter);
+                          $parameter = str_replace($match, '\''.addcslashes($node->nodeValue, "'").'\'', $parameter);
```
Comment (djjuhasz) [inline lib/QubitXmlImport.class.php:661]: As far as I can tell, this allows arbitrary execution (by PHP) of *any* string used for `$node->nodeValue` in line 671 below. :(

**Review by djjuhasz — CHANGES_REQUESTED** [review]
I think the `eval()` statement in line 671 needs to be removed altogether, as it's a security flaw. :(

File: lib/QubitXmlImport.class.php (hunk 1 above)
Comment (mcantelon) [reply lib/QubitXmlImport.class.php:661]: True. I think most of our XML importing relies on this mechanism, which is not so good.

Comment (sbreker) [reply lib/QubitXmlImport.class.php:661]: @mcantelon will be creating a new ticket describing the security issue with the eval() use.

Comment (sbreker) [reply lib/QubitXmlImport.class.php:661]: Ticket describing eval() issue:
https://projects.artefactual.com/issues/13216

**Review by sbreker — APPROVED** [review]
Looks good @mcantelon. Thanks for creating that ticket!

**Review by djjuhasz — APPROVED** [review]
Looks good 👍

**Comment by mcantelon** [comment]
Thanks @sbreker !

**Pull request merged**